# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1368225064 14400
#      Fri May 10 18:31:04 2013 -0400
# Node ID d788ee8b8a99f59dad09ecfc8a2570621e99111a
# Parent  25e418d23a4b00750f38e46b7f7d5e4e5352afa0
Fix GUI includes when built without Qscintilla

* settings-dialog.cc: Unconditionally include color-picker.h. Do not
include QSettings here.
* settings-dialog.h: Include QSettings here.

diff --git a/libgui/src/settings-dialog.cc b/libgui/src/settings-dialog.cc
--- a/libgui/src/settings-dialog.cc
+++ b/libgui/src/settings-dialog.cc
@@ -19,28 +19,27 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
+#include "color-picker.h"
 #include "resource-manager.h"
 #include "workspace-model.h"
 #include "settings-dialog.h"
 #include "ui-settings-dialog.h"
-#include <QSettings>
 #include <QDir>
 #include <QFileInfo>
 #include <QVector>
 
 #ifdef HAVE_QSCINTILLA
 #include <QScrollArea>
-#include "color-picker.h"
 #include <Qsci/qscilexercpp.h>
 #include <Qsci/qscilexerbash.h>
 #include <Qsci/qscilexerperl.h>
 #include <Qsci/qscilexerbatch.h>
 #include <Qsci/qscilexerdiff.h>
 #endif
 
 settings_dialog::settings_dialog (QWidget *p):
diff --git a/libgui/src/settings-dialog.h b/libgui/src/settings-dialog.h
--- a/libgui/src/settings-dialog.h
+++ b/libgui/src/settings-dialog.h
@@ -19,16 +19,17 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifndef SETTINGSDIALOG_H
 #define SETTINGSDIALOG_H
 
 #include <QDialog>
+#include <QSettings>
 #ifdef HAVE_QSCINTILLA
 #include "lexer-octave-gui.h"
 #endif
 
 namespace Ui
 {
   class settings_dialog;
 }
